src/main/java/mate/academy/dao/ShoppingCartDao.java

```diff
@@ -0,0 +1,13 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+
+public interface ShoppingCartDao {
+    ShoppingCart add(ShoppingCart shoppingCart);
+
+    Optional<ShoppingCart> getByUser(User user);
```

Comment:
```suggestion
    Optional<ShoppingCart> findByUser(User user);
```
if you return `Optional` let's name method as `findBy`